[xiaorui-ui] iP

xiaorui-ui opened the pull request:
# Dune

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Dune frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ **SUPER FAST** to use

All you need to do is,

1. download it from [here](https://github.com/xiaorui-ui/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is *FREE*!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon<sup>TM</sup>)
- [ ] Reminders (coming soon<sup>TM</sup>)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

xiaorui-ui committed:
level 0

xiaorui-ui committed:
level 1

xiaorui-ui committed:
level-2

xiaorui-ui committed:
level 3

xiaorui-ui committed:
level 4

xiaorui-ui committed:
Abstraction for adding all types of events

xiaorui-ui committed:
text ui testing

xiaorui-ui committed:
level-5

xiaorui-ui committed:
level-6

xiaorui-ui committed:
Level-7

Handled case when the file exists in the specified path

xiaorui-ui committed:
Fix bugs
I spent so long on this...

xiaorui-ui committed:
Create file on start-up(and parent directory) if absent

xiaorui-ui committed:
Add dateTime functionalities

xiaorui-ui committed:
Merge branch 'branch-Level-8'

Finish level 8

xiaorui-ui committed:
Create Ui and Storage classes

xiaorui-ui committed:
Create Parser and TaskList classes

xiaorui-ui committed:
Merge branch 'A-MoreOOP'

Implement OOP

xiaorui-ui committed:
Create packages

xiaorui-ui committed:
Merge branch 'A-Packages'

Add packages

xiaorui-ui committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Add Gradle Support for current project

xiaorui-ui committed:
Set up Gradle

xiaorui-ui committed:
Add event test

xiaorui-ui committed:
Add storagetest

xiaorui-ui committed:
Merge branch 'A-JUnit'

Add test cases to master

xiaorui-ui committed:
Build jar file

xiaorui-ui committed:
Clean up code

xiaorui-ui committed:
Create Javadocs

xiaorui-ui committed:
Merge branch 'branch-A-Javadoc'

Create Javadoc

xiaorui-ui committed:
Follow coding standards

xiaorui-ui committed:
Merge branch 'branch-A-CodingStandard'

Resolve conflicts in Event class.

# Conflicts:
#	src/main/java/dune/task/Event.java

xiaorui-ui committed:
Implement Level-9

Add in functionality to search for tasks via key words

xiaorui-ui committed:
Merge branch 'branch-Level-9'

Resolve conflicts in TaskList.

# Conflicts:
#	src/main/java/dune/task/TaskList.java

xiaorui-ui committed:
Set up JavaFX

GERARDJM018 commented on src/main/java/dune/task/TaskList.java:
```diff
@@ -0,0 +1,190 @@
+public class TaskList {
+    private List<Task> tasks;
+
+    private static String before = "Start date must be before end date";
```
Perhaps use a name that could explain the usage of the variable (e.g. = dateErrorMessage)?

GERARDJM018 commented on src/main/java/dune/task/TaskList.java:
```diff
@@ -0,0 +1,190 @@
+     * @param storage  Storage object to save tasks.
+     */
+    public void addTask(int i, String text, Storage storage) {
+        Task x = null;
```
Perhaps use a more detailed name (e.g. = taskToBeAdded) (since the variable is used throughout the method (not just used on a few lines)?

GERARDJM018 commented on src/main/java/dune/Storage.java:
```diff
@@ -0,0 +1,154 @@
+        // System.out.println(eventType);
+        boolean isDone = (components[1].equals("1")) ? true: false;
+        if (eventType.equals("T")) {
+            return new ToDo(components[2], isDone);
+        } else if (eventType.equals("D")) {
+            return new Deadline(components[2], components[3], isDone);
+        } else if (eventType.equals("E")) {
+            return new Event(components[2], components[3], components[4], isDone);
+        } else {
+            // System.out.println("Dune exception!");
+            throw new DuneException("Invalid task type");
+        }
+    }
+
+    // Dates are in the format yyyy-mm-ddTHH:MM, unlike what's printed
+
+    /**
+     * Converts a task to a line to be written to the file.
+     *
+     * @param t A task.
+     * @return
+     * @throws DuneException
+     */
+    public String convertTaskToLine(Task t) throws DuneException {
+        String ans = "";
+        if (t instanceof ToDo) {
+            ans = "T|" + (t.getIsDone() ? "1" : "0") + "|" + t.getDescription();
+        } else if (t instanceof Deadline) {
+            Deadline d = (Deadline) t;
+            ans = "D|" + (t.getIsDone() ? "1" : "0") + "|" + t.getDescription() + "|"
+                    + d.getDeadline();
+        } else if (t instanceof Event) {
+            Event e = (Event) t;
+            ans = "E|" + (t.getIsDone() ? "1" : "0") + "|" + t.getDescription() + "|"
+                    + e.getStart() + "|" + e.getEnd();
+        } else {
+            throw new DuneException("Invalid task type");
+        }
+        return ans;
+    }
```
I like the naming of these methods. It explains what each method does.

GERARDJM018 commented on src/main/java/dune/Dune.java:
```diff
@@ -0,0 +1,48 @@
+
+    private Storage storage;
+
+    private Parser p;
```
Perhaps use a more detailed name here (e.g. = parser)? Since it is a global variable for the class which will be used throughout the class.

GERARDJM018 commented on src/main/java/dune/DateTimePrinter.java:
```diff
@@ -0,0 +1,37 @@
+public class DateTimePrinter {
+
+    /** Months in the year. */
+    private String[] months = new String[] {
```
Perhaps use a more detailed name here (e.g. = listOfMonths, nameOfMonths)? Because it might cause some misleading like thinking it of a variable used to store months used.

GERARDJM018 reviewed (commented):
Overall great use of naming for variables and methods. 👍 
Some names could be more detailed or changed (added some suggestions 😉).

xiaorui-ui committed:
Set up Javafx

Create a fully functional GUI, step 3 of tutorial

xiaorui-ui committed:
Set up Java FX

Level 4

xiaorui-ui commented:
@GERARDJM018 noted, thanks!

xiaorui-ui committed:
Set up GUI

xiaorui-ui committed:
Merge branch 'Level-10'

Resolve conflicts

xiaorui-ui committed:
Fix minor changes

xiaorui-ui committed:
Create dummy assertion

No assertion statements to check that the program
is working correctly. So I added a dummy one that is
always false to check assertions are activated.

xiaorui-ui committed:
Include more assertions

Ensure correctness in the Storage class thoroughly.

xiaorui-ui committed:
Improve code quality

*broke up long methods
*reduce nested code

xiaorui-ui committed:
Merge pull request #1 from xiaorui-ui/branch-A-Assertions

Branch a assertions

xiaorui-ui committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	data/dune.txt
#	src/main/java/dune/Parser.java
#	src/main/java/dune/Storage.java

xiaorui-ui committed:
Merge pull request #2 from xiaorui-ui/branch-A-CodeQuality

Branch a code quality

xiaorui-ui committed:
Add support for duplicates

xiaorui-ui committed:
Add more support for duplicates

xiaorui-ui committed:
Merge branch 'C-DetectDuplicates'

xiaorui-ui committed:
Improve code quality

xiaorui-ui committed:
Create user guide

xiaorui-ui committed:
Update user guide

xiaorui-ui committed:
Update README

xiaorui-ui committed:
Update README

xiaorui-ui committed:
Clean up code

xiaorui-ui committed:
Disable empty task description

xiaorui-ui committed:
Stop line truncation in GUI

xiaorui-ui committed:
Update unit tests

xiaorui-ui committed:
Update user guide

xiaorui-ui committed:
Check coding style
